feature/Fine-Grained Admin Permissions (FGAP) V2 compatibility with Keycloak 26.2.0+

AssahBismarkabah opened the pull request:
### **What this PR does / why we need it**:

This PR fixes Fine-Grained Admin Permissions (FGAP) V2 compatibility issues with Keycloak 26.2.0+ by implementing graceful error handling for HTTP 501 responses and maintaining backward compatibility with older Keycloak versions.

### **Key Changes:**
- Added `@DisabledIfSystemProperty` annotations to disable FGAP V1 tests for Keycloak 26.2.0+
- Implemented graceful error handling in `ClientRepository` for HTTP 501 responses
- Updated version-specific baseline files for Keycloak 26.3.3 compatibility
- Fixed recursive property references in [`pom.xml`](https://github.com/adorsys/keycloak-config-cli/blob/9b4f713cbc93c79bff298d48ae49fd50ed609839/pom.xml#L1083C1-L1098C19)
- Added comprehensive test suite for FGAP V2 compatibility validation

###**Which issue this PR fixes**: 
Fixes #1305  - HTTP 501 Not Implemented with fine-grained permissions on Keycloak 26.3.0

### **Special notes for your reviewer**:

The core issue was not that the CLI uses V1 APIs (it correctly uses V2 APIs), but that FGAP V2 must be enabled per realm even when enabled at the server level. The HTTP 501 error occurs when FGAP V2 is disabled for a specific realm, which is expected behavior.

### **Testing:**
- All tests pass with Keycloak 26.3.3 (`mvn clean install -Dkeycloak.version=26.3.3`)
- FGAP V1 tests are properly disabled for 26.2.0+ versions
- Backward compatibility maintained for Keycloak ≤ 26.1.x
- HTTP 501 errors are handled gracefully with informative warnings

### **PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [x] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

AssahBismarkabah commented:
> @AssahBismarkabah maybe why you test are failing check contrib/custom-representations/src/main/java/org/keycloak/representations/idm/RealmRepresentation.java then public void setLoginTheme(String loginTheme) { this.loginTheme = loginTheme; }
> 
> change to
> 
> public void setLoginTheme(String loginTheme) { // prevent DB constraint violation (VARCHAR 255) if (loginTheme != null && loginTheme.length() > 255) { loginTheme = loginTheme.substring(0, 255); } this.loginTheme = loginTheme; }
> 
> then see if test are green then

Hello @jwklijnsma, thanks for the proposal, 
 Changing `contrib/custom-representations/src/main/java/org/keycloak/representations/idm/RealmRepresentation.java#setLoginTheme` will only take effect if that custom representation class is actually on the runtime classpath (or the server loads the custom JAR). In the current test/runtime setup it will not be used automatically , so editing that file alone will not fix the IT test failure

i am currently implementing a client-side validation in RealmRepository so bad input causes an explicit exception in the CLI before the request reaches Keycloak.

AssahBismarkabah commented on src/main/java/de/adorsys/keycloak/config/service/ClientAuthorizationImportService.java:
```diff
@@ -66,6 +70,17 @@ public class ClientAuthorizationImportService {
+    public static final String ADMIN_PERMISSIONS_CLIENT_ID = "admin-permissions";
+    private static final int HTTP_NOT_FOUND = 404;
+    private static final int HTTP_NOT_IMPLEMENTED = 501;
+    private static final String FGAP_V2_WARNING = "Authorization API not supported (likely FGAP V2 active). "
```
@thomasdarimont thanks for the review! I have implemented a definitive FGAP V2 detection using the `/admin/serverinfo` endpoint as you suggested. Also addressed the attempt update to the system-managed admin-permissions client, can you test again on your side ?
you can verify with https://github.com/adorsys/keycloak-config-cli/blob/fix-issue-1305-FGAP-returns-501-for-keycloak-26.2.0+/contrib/example-config/fgap-v2.json
<img width="1840" height="1195" alt="Screenshot 2025-10-30 at 17 34 23" src="https://github.com/user-attachments/assets/d512e5cd-ebf9-4e08-86b2-200c89951c23" />

antikalk replied:
@AssahBismarkabah Thanks, this looks great! So the initial import is working fine - but I've noticed the same issue as @rpion05 on update of a policy. There might as well be something that does not work the same way as it did with FGAP v1...

antikalk commented:
> Authorization policies must then be managed post-import through the Admin Console or REST API, not through declarative configuration files.

imo it would still be great to be able to configure the policies with config-cli - to automate and ensure their configuration, instead of manually creating them which could lead to mistakes. 

Unfortunately it looks like the migration after all is not as easy as initially thought...

AssahBismarkabah commented:
Hello @rpion05 @antikalk  

I've implemented the changes to resolve the FGAP V2 409 Conflict issue.
As we suspected, the root cause was Keycloak blocking API access to the admin-permissions client, so I've updated the logic to explicitly skip authorization settings for that specific client when FGAP V2 is active, which aligns with Keycloak's own import behavior. 
I also updated the documentation and example config to clarify that while system-level admin permissions must be managed manually, you can still fully configure authorization for your own application clients. Please give it another try and let me know if you run into any issues.

jwklijnsma commented:
@AssahBismarkabah maybe get more maintainer for project maybe helps ?
